### PR TITLE
d_do_test: Add missing DDOC comments

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
Because every declaration should have a short description of its purpose (and parameters).

P.S.: Wrote down some mental notes about the test runner, therefore some comments can probably be improved (suggestions are welcome).

### CODE REVIEW

File: test/tools/d_do_test.d [hunk 1]
```diff
@@ -264,6 +313,25 @@ bool findTestParameter(const ref EnvData envData, string file, string token, ref
     return true;
 }
 
+/**
+ * Read the multi-line test parameter `token` from the source code and joins
+ * multiple definitions into a single string.
+ *
+ * ---
+ * TEST_OUTPUT:
+ * --
```
Comment (MoonlightSentinel) [inline test/tools/d_do_test.d:322]: How to escape `---` inside of a code block?

File: test/tools/d_do_test.d [hunk 2]
```diff
@@ -726,6 +836,22 @@ unittest
...
+ * and appends the generated objects to `sources`.
+ *
+ * Params:
+ *   input_dir    = test directory (e.g. `runnable`)
+ *   output_dir   = directory for intermediate files
+ *   extraSources = sources to compile
+ *   sources      = list of D sources to extend with object files
+ *   envData      = environment configuration
+ *   compiler     = external compiler (E.g. clang)
+ *   cxxflags     = compiler flags
```
Comment (wilzbach) [inline test/tools/d_do_test.d:850]:
```suggestion
 *   cxxflags     = external compiler flags
```

File: test/tools/d_do_test.d [hunk 3]
```diff
@@ -1597,6 +1731,16 @@ int tryMain(string[] args)
     return 0;
 }
 
+/**
+ * Executes a bash script (deprecated).
```
Comment (wilzbach) [inline test/tools/d_do_test.d:1735]:
```suggestion
 * Executes a bash script (deprecated). Postprocessors or `dshell` can be used instead.
```

**Review by wilzbach — APPROVED** [review]
This is great!

File: test/tools/d_do_test.d [hunk 4]
```diff
@@ -305,13 +373,16 @@ bool findOutputParameter(string file, string token, out string result, string se
     return found;
 }
 
+/// Replaces the placeholer `${RESULTS_DIR}` with the actual path
+///  to `test_results` stored in `envData`.
```
[the remark this answers is not in the document]
Comment (MoonlightSentinel) [reply test/tools/d_do_test.d]: No, just to much documentation for one evening.

File: test/tools/d_do_test.d (hunk 3 above)
Comment (MoonlightSentinel) [reply test/tools/d_do_test.d]: Amended to mention `dshell` test as `POST_SCRIPT` is up for deprecation as well

File: test/tools/d_do_test.d (hunk 1 above)
Comment (MoonlightSentinel) [reply test/tools/d_do_test.d]: Done, allthough it does require `preview=markdown` (and isn't rendered correctly in code-d...)

**Comment by NilsLankila** [comment]
No reason to wait but the [ddoc is not generated](http://dtest.dlang.io/results/955fb405b5f3da8109c5398dde8623ebc2715d5d/cb2f3ab033b5241e85e986e0239a9f95edc1e774/), are you aware of that ? I guess it's only supposed to be used e.g with DCD then ?

**Comment by MoonlightSentinel** [comment]
Yes, but it's still available in IDE's (e.g. hovering works fine in `code-d`).

[Not to mention that the documentation was severily lacking - ddoc or not.]

**Review by ghost — APPROVED** [review]

**Pull request merged**